### ISSUE TITLE
In-Game: [HTML5] surface_resize() with application_surface only applies the resize on the first request

### ISSUE THREAD

**Issue by Foib** [YoYoGames/GameMaker-Bugs#7666]
### Description

When `surface_resize(application_surface, w, h)` is called on the HTML5 platform, the size of the surface is only changed once. After that, another call does not change the size of the surface.

The resizing of manually created surfaces works.

Resizing the application surface on the Windows platform works.

[bugPackage.zip](https://github.com/user-attachments/files/17016059/bugPackage.zip)

https://github.com/user-attachments/assets/ebc6fb16-95b6-417e-a33c-260a78e95815

### Expected Change

Size of the application surface changes when calling surface_resize on HTML5 target

### Steps To Reproduce

1. Start GameMaker
2. Set HTML5 platform
3. Create object
4. In `Alarm 0` event write `surface_resize(application_surface, 5, 10)`
5. In `Alarm 1` event write `surface_resize(application_surface, 15, 20)`
6. In `Create` event write `alarm[0] = 1; alarm[1] = 2`
7. In `Draw` event write `show_debug_message($"{surface_get_width(application_surface)},{surface_get_height(application_surface)}")`
8. Place object inside room
9. Run game and open browser dev tools
10. The console output should now be `5,10`

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.0 (Monthly)

### Which platform(s) are you seeing the problem on?

HTML5

### Sample Package Attached?

- [X] I have attached my bug package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by guillermoares** [comment]
I'm experiencing the same issue. 

In case it is of any help, this wasn't happening 28 days ago, so this was probably introduced in the last updates (there were 2 since then, right?).

**Comment by guillermoares** [comment]
I just confirmed this issue is not present in version `IDE v2024.6.2.162  Runtime v2024.6.1.208`. 

In case someone wants to go back, it can be downloaded from here:
https://releases.gamemaker.io/release-notes/2024/6

There was a tileset format update since then which is not backwards compatible. I was able to get the format migration undone by installing the older version and playing a bit with the IDE. I didn't quite understand what finally triggered the "de-migration" though. It was something around opening the tiles, closing them again, closing the IDE, reopening it, etc until it happened.

**Comment by Retrific** [comment]
I am also experiencing this issue

**Comment by guillermoares** [comment]
Thanks for the quick reply! I didn't mean to disturb your Sunday 😅 

It would be nice to be able to disable all these checks globally in release builds. I just tried a few things and it looks like it is possible to do something like this:

```javascript
#macro PRAGMA_JS_OPTIMIZE gml_pragma("optimise", "js_array_check", "push, off"); gml_pragma("optimise", "js_error_check", "push, off"); gml_pragma("optimise", "js_check_index", "push, off"); gml_pragma("optimise", "js_pre_post_no_long", "push, on"); gml_pragma("optimise", "js_use_infix_ops", "push, on")
```

And then run a script like this one to add `PRAGMA_JS_OPTIMIZATION` at the beginning of all files:

```cmd
@echo off
setlocal enabledelayedexpansion

:: Search for .gml files recursively
for /r %%f in (*.gml) do (
    :: Check if the file contains 'PRAGMA_JS_OPTIMIZE'
    findstr /M /C:"JS_OPTIMIZE" "%%f" >nul
    if errorlevel 1 (
        :: If not found, prepend 'PRAGMA_JS_OPTIMIZE'
        echo Prepending JS_OPTIMIZE to %%f
        echo JS_OPTIMIZE > temp.txt
        type "%%f" >> temp.txt
        move /Y temp.txt "%%f" >nul
    )
)
```
Credit for this script goes entirely to ChatGPT 🙌 It'll look up all GML files and add the macro if it's not already there.

Then you can set `PRAGMA_JS_OPTIMIZATION` to nothing (`#macro PRAGMA_JS_OPTIMIZATION // nothing` worked for me) when you want the fail-safe JS stuff in, and back to the `gml_pragma` calls otherwise. I'm building my game with a script using Igor calls, so it shouldn't be too hard to automatically set the macro right before building and resetting it back right after.

I also found a curious thing: It seems arithmetic operations fail to recognize parenthesys when `js_use_infix_ops` is off. I had something like `(a - b) / 2`, which got compiled into `a - b / 2`, which effectively resulted in `a - (b / 2)`. I might make a proper report for this later. Actually, my failing example was `(GUI_WIDTH - _width) / 2`, where `GUI_WIDTH` is a macro for `display_get_gui_width()` and `_width` is a local variable (`var _width`). Maybe the macro had something to do with it(?).

**Comment by cameron-home** [comment]
Verified fixed in IDE v2024.1100.0.639  Runtime v2024.1100.0.662
